src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -1039,30 +1076,49 @@ public Tensor contiguous()
                     if (idx.kind == TensorIndex.Kind.Tensor && (object?)idx.tensor != null)
                         hasTensor = true;
                 }
+
                 arrStops = hasSliceEnd ? new long[n] : null;
                 arrSteps = hasSliceStep ? new long[n] : null;
                 arrTensors = hasTensor ? new IntPtr[n] : null;
                 arrKindAndStarts = new long[n];
                 for (int i = 0; i < indices.Length; i++) {
                     var idx = indices[i];
                     arrKindAndStarts[i] =
-                        (idx.kind == TensorIndex.Kind.Null) ? long.MinValue :
-                        (idx.kind == TensorIndex.Kind.Bool && idx.startIndexOrBoolOrSingle == 0) ? long.MinValue + 1 :
-                        (idx.kind == TensorIndex.Kind.Bool && idx.startIndexOrBoolOrSingle != 0) ? long.MinValue + 2 :
-                        (idx.kind == TensorIndex.Kind.Ellipsis) ? long.MinValue + 3 :
-                        (idx.kind == TensorIndex.Kind.None) ? long.MinValue + 4 :
-                        (idx.kind == TensorIndex.Kind.Tensor) ? long.MinValue + 5 :
-                        (idx.kind == TensorIndex.Kind.Slice && !idx.startIndexOrBoolOrSingle.HasValue) ? long.MinValue + 6 :
-                        (idx.kind == TensorIndex.Kind.Single) ? idx.startIndexOrBoolOrSingle.GetValueOrDefault() :
-                        idx.startIndexOrBoolOrSingle.GetValueOrDefault() + long.MinValue / 2;
+                        (idx.kind == TensorIndex.Kind.Null)
```

Comment:
Was this the result of some automatic re-formatting? I prefer the original. In general, I would suggest against PRs with reformatting, since it makes the diff much harder to read and spot the actual logical changes. Unless it's egregiously bad in the original, of course.

Same comment applies to all the re-formatting edits.

Reply:
Fixed